src/types/types.ts

```diff
@@ -0,0 +1,25 @@
+import { store } from "../app/store";
+
+export type AppDispatch = typeof store.dispatch;
+
+export type UserReturnState = {
+    data: {
+      user: InitialState;
+    }
+  }
+  
+  export type UserDetails = {
+    uid?: string
+    userName: string | null
+    email?: string | null
+  }
+  
+  export type InitialState = {
+    user: UserDetails
+    isLoading: Boolean
+  }
+  
+  export const initialState: InitialState = {
+    user: {userName: ""},
+    isLoading: true,
+  };
```

Comment:
Improve formatting or start using eslint :P
btw we've created `/types/types.ts` path - what will be other files in this directory? does `types.ts` for that file make sense? It there will be no more "type" so what is purpose of that dir? Or maybe we should name it "user.ts" as it contains all the types related to user related things hm?